corehq/apps/userreports/tasks.py

```diff
@@ -306,9 +306,10 @@ def reprocess_archive_stubs():
         # If the history wasn't updated the first time around, run the whole thing again.
         if not stub.history_updated:
             if stub.archive:
-                xform.archive(user_id=stub.user_id)
+                FormAccessors.do_archive(xform, True, stub.user_id, trigger_signals=True)
```

Comment:
Yeah that's right.

@snopoke the long run time of `reprocess_archive_stubs` is because of `UnfinishedArchiveStub` being permanently in the queue, because processing them does not actually delete them, because (as Daniel said) `xform.archive(user_id=stub.user_id)` does nothing if `xform` is already archived, and thus doesn't trigger the steps that delete the stub.

Because we don't know when exactly it failed—all we know is it was after setting `xform.is_archived` in the db—the safest thing seems to redo the whole archive process, provided that we're sure it's idempotent.

The alternative would be to assume that everything went well and just retrigger signals or something like that and delete the stub. This seems more likely to leave some things missed, but given no one's complained about these forms I'm not sure how big an issue that is, it just seems more likely to have a "leak"